pgex/examples/platformer_collisions/main.py

```diff
@@ -0,0 +1,125 @@
+"""
+This file is a part of the 'Pygame Examples (pgex)' source code.
+The source code is distributed under the MIT license.
+"""
+
+import asyncio
+
+import pygame
+
+from . import common
+
+
+def get_colliding_tiles(tiles: list, player_rect: pygame.Rect) -> list:
```

Comment:
Is it intended to always use the generic type `list` instead of specifying what is the type of the values inside the given list?

For example `list[Tile]`, it could help readers having more insight about what data is manipulated and let the IDE give hints while developing.

Reply:
Considering we are using 3.8 (or 3.7) compatible type hints it should be `t.List[Tile]` or `List[Tile]` but yeah, agreed, on the other hand though, people have asked about what those even mean because as beginners they may have not seen type hints very often if at all, but we have them because a good example is a good example. But yeah, I agree with the overall idea.

Reply:
I forgot it is a really recent feature, makes sense to support older versions. 

Type hints is something ok for beginners I think. 
Almost all regular languages are strongly typed anyway, even JavaScript: most people are using TypeScript now.